blocks/library/shortcode/block.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId, Dashicon } from '@wordpress/components';
+import { Component, compose } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import ShortcodePreview from './preview';
+import BlockControls from '../../block-controls';
+import { getCurrentPostId } from '../../../editor/store/selectors';
+import PlainText from '../../plain-text';
+
+export class Shortcode extends Component {
+	constructor() {
+		super();
+		this.state = {
+			preview: false,
+		};
+	}
+
+	render() {
+		const { preview } = this.state;
+		const { instanceId, postId, setAttributes, attributes, focus, setFocus } = this.props;
+		const inputId = `blocks-shortcode-input-${ instanceId }`;
+
+		const controls = focus && (
+			<BlockControls key="controls">
+				<div className="components-toolbar">
+					<button
+						className={ `components-tab-button ${ ! preview ? 'is-active' : '' }` }
+						onClick={ () => this.setState( { preview: false } ) }>
+						<span>{ __( 'Shortcode' ) }</span>
+					</button>
+					<button
+						className={ `components-tab-button ${ preview ? 'is-active' : '' }` }
+						onClick={ () => this.setState( { preview: true } ) }>
+						<span>{ __( 'Preview' ) }</span>
+					</button>
+				</div>
+			</BlockControls>
+		);
+
+		if ( ! preview ) {
```

Comment:
Can we flip this check and put code responsible for the preview in here?

```js
if ( preview ) {
	return [
		controls,
		<ShortcodePreview key="preview"
			shortcode={ shortcodeContent }
			setFocus={ setFocus }
		/>,
	];
}

// code continues with the default behavior
```

Reply:
Yes, flipping it is possible. It is cleaner to have the positive condition appear first, than the negative condition, and I was on two minds if I should have the positive condition appear first. However, I went with keeping the negative condition (ie ! preview) first since it will show the UI that appears first when one loads the editor. Happy to change it per your suggestion.